### PR TITLE
GEN-1377-coverage

### PR DESCRIPTION
**Pull request by fedealconada**
## Motivation

Have test coverage so we know where we are standing at with regards to tests, which ones are missing and where.

## Solution

- [x] fix remappings so `forge coverage` works well
- [x] add codecov and badge to README  
- [x] remove `Trust.sol` that we were not using
- [x] ignore contracts that don't need to be covered
 
## Implementer Checklist

- [ ]  [OPTIONAL] Create a reference implementation (in python or JS)
- [x]  Create a short "Motivation" section for the PR
- [x]  Write a spec for the feature and put it in the PR description – basic function names and expected state transitions is OK
- [ ]  Get the PR reviewed by at least two people

If this is your first time reviewing smart contract changes
- [ ]  Review the [Solcurity](https://github.com/Rari-Capital/solcurity) standard

If smart contract changes were made
- [ ]  Check all of the new revert paths with concrete tests
- [ ]  Check all of the new storage slot writes with concrete tests – do not make the passing test case the trivial case!
- [ ]  Go line-by-line through the new code and ensure it has all been covered by concrete tests – since we don’t have a coverage engine for foundry yet, you can imagine how one might check that each different code path is covered
- [ ]  Simplify the implementation and spend some time trying to minimize gas costs
- [ ]  Write fuzz tests for the feature – try everything to break the implementation (is this function monotonically in/decreasing, should it always be less than something else, etc)
- [ ]  Capture bugs discovered during the above steps in concrete tests (regression tests)
- [ ]  Add integration tests – fuzz or concrete tests that test how the new code affects the entire system, either locally or with a mainnet fork
- [ ]  [OPTIONAL] Integrate the feature into the deployment scripts
- [ ]  [OPTIONAL] Add new sanity checks to the deployment scripts

If your PR uses or interacts with prices of any kind
- [ ] Make sure forge is running the test on the latest block
- [ ] Compare the value returned with some other source (like Curve)
- [ ] Check the `updatedAt` (if available) value of the contract and confirm that the price we are sourcing is in fact not older than X
- [ ] Ensure oracle liveness from the oracle maintainer, either through a public status interface or through written confirmation from a developer.

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# interfaces are automatically ignored by forge coverage
+ignore:
+  - "**/tests/**/*"                         # ignore all files inside tests/ folders
```
Comment (jparklev) [inline codecov.yml:3]: Nice one, we should add this to the auto roller as well

File: pkg/core/.gas-snapshot [hunk 2]
```diff
@@ -1,154 +1,319 @@
...
-PeripheryTest:testAdminVerifyAdapterWhenPermissionless() (gas: 3460413)
-PeripheryTest:testCantDeployAdapterIfTargetIsNotSupported() (gas: 1408212)
-PeripheryTest:testCantDeployAdapterIfTargetIsNotSupportedOnSpecificAdapter() (gas: 7088783)
-PeripheryTest:testCantMigrateLiquidityIfTargetsAreDifferent() (gas: 7168805)
-PeripheryTest:testCantVerifyAdapterNonAdmin() (gas: 3285407)
-PeripheryTest:testCantVerifyAdapterNonAdminWhenPermissionless() (gas: 3297226)
-PeripheryTest:testDeployAdapter() (gas: 3652687)
-PeripheryTest:testDeployAdapterWhenPermissionless() (gas: 3659960)
-PeripheryTest:testDeployPeriphery() (gas: 4236538)
+DateTimeTest:testFormatEleventhAndMay() (gas: 14054)
```
Comment (jparklev) [inline pkg/core/.gas-snapshot]: I wonder if it's time to retire these gas snapshots

**Review by jparklev — COMMENTED** [review]
I see that CI is passing with the new remappings! Good stuff.

I'm currently seeing this "coverage data is unkown" message on codecov, is there anything I need to do to see the breakdown?
Screen Shot 2022-11-03 at 10 31 33 AM

File: pkg/core/.gas-snapshot (hunk 2 above)
Comment (fedealconada) [reply pkg/core/.gas-snapshot]: you mean removing the CI.yml step (as well as the files)?

File: codecov.yml (hunk 1 above)
Comment (fedealconada) [reply codecov.yml:3]: I'll add it

File: pkg/core/.gas-snapshot (hunk 2 above)
Comment (jparklev) [reply pkg/core/.gas-snapshot]: yea!

Comment (fedealconada) [reply pkg/core/.gas-snapshot]: sure! i'll remove it

**Review by jparklev — APPROVED** [review]
Confirmed w fede that codecov is looking good

**Review by abylaw — APPROVED** [review]
nice!! curious why we need to move things out of `src`?

**Comment by fedealconada** [comment]
> nice!! curious why we need to move things out of `src`?

haha, not sure exactly why but that was making `forge coverage` to run, otherwise, it was throwing an error. If you add the `/src` back, and remove it from the `foundry.toml` files, throws the error 🤷🏼‍♂️

**Pull request merged**